### DISCUSSION TITLE
Question on fit_regress

### DISCUSSION THREAD

**Discussion by florapython** [rsagroup/rsatoolbox#397] [Q&A] [answered]
Hello all

First thank you for a great toolbox! 

I have a question on model.fit_regress. If I understand correctly by reading the class Fitter in the code, this function first transforms the data into a data_mean = pool_rdm(data, method=method). Am I correct? So there is no way to keep a set of RDMs intact (for example, the result of "get_searchlight_RDMs" with many RDMs for searchlight spheres) and run a regression for each searchlight sphere independently to get an array of thetas for each searchlight sphere?
If not, I'll do a for loop.

I have another related question on evaluate_models_searchlight. If I don't give any thetas to the function, but I give it a weighted model, it supposes that thetas are all ones? Is there a way to tell this function to first do a linear regression on the weighted model before evaluating it? Just to avoid the for loop mentioned above.

Thanks!

**Comment by JasperVanDenBosch** [comment]
You're very welcome and thanks for contributing to the conversation here!

Yes, your understanding of `fit_regress` is correct, for a repeated analysis you'd have to call fit for every subset separately. 

However, the crossvalidation-based evautation functions include a fitting step. For say `eval_fixed` you are right and the `theta`'s would be ones (as it is meant for fixed models). But if you use the `eval_dual_bootstrap` evaluation, the thetas are fit as part of the process. 

So you could do `evaluate_models_searchlight(rdm, model_rdm, eval_dual_bootstrap, method="cosine")` for instance.

Unfortunately the searchlight code is currently not actively maintained. We have plans to optimize the searchlight (and "repeated" RSA more generally) this over the next few months.

**Reply by florapython** [reply]
thank you!

**Reply by JasperVanDenBosch** [reply, marked as answer]
When you've tried this out and it works, can you "Mark" this answer as the correct one? (For future visitors to find their way 🙏🏻)
